api/config/v1alpha1/envoygateway_metric_types.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+type EnvoyGatewayMetrics struct {
+	// Sinks defines the metric sinks where metrics are sent to.
+	Sinks []EnvoyGatewayMetricSink `json:"sinks,omitempty"`
+	// Matches defines configuration for selecting specific metrics instead of generating all metrics stats
+	// that are enabled by default. This helps reduce CPU and memory overhead in Envoy Gateway.
+	Matches []Match `json:"matches,omitempty"`
```

Comment:
how will these work?
how to support Prometheus? EG should always support prometheus first.